src/adv.py

```diff
@@ -49,3 +88,23 @@
 # Print an error message if the movement isn't allowed.
 #
 # If the user enters "q", quit the game.
+
+print('Enter "n" for North, "s" for South, "e" for East, "w" for West and "q" to quit the game.')
+while True:
+    print(first_player)
+    player_input = input("-> ")
+
+    if player_input in possible_inputs:
+        # result = check_bad_input(first_player, player_input)
+        if player_input == 'q':
+            print(f"Goodbye {first_player.name}")
+            break
+        if not advance_room(first_player, player_input):
+            print('You shall not pass! Read the description and choose your path carefully!')
+    else:
+        print('Invalid command! \nPlease enter "n" for North, "s" for South, "e" for East, "w" for West and "q" to quit the game.')
+
```

Comment:
Good job formatting logs. It's a simple thing but can help both you and other devs in the future.